src/main/java/com/amihaiemil/docker/RtContainers.java

```diff
@@ -109,20 +105,21 @@ public Container create(
         post.setEntity(new StringEntity(container.toString()));
         post.setHeader(new BasicHeader("Content-Type", "application/json"));
         final HttpResponse response = this.client.execute(post);
-        final Container created;
-        if(response.getStatusLine().getStatusCode() == HttpStatus.SC_CREATED) {
+        final int status = response.getStatusLine().getStatusCode();
+        if(status == HttpStatus.SC_CREATED) {
             final JsonObject json = Json
                 .createReader(response.getEntity().getContent()).readObject();
-            created = new RtContainer(
+            post.releaseConnection();
+            return new RtContainer(
                 this.client,
                 URI.create(
                     this.baseUri.toString() + "/" + json.getString("Id")
                 )
             );
-        } else {
-            created = null;
         }
-        return created;
+        throw new UnexpectedResponseException(
+            "Containers#create(...)", status, HttpStatus.SC_CREATED
```

Comment:
@amihaiemil the first argument to ctor of `UnexpectedResponseException` is supposed to be the endpoint

Reply:
@llorllale Yes, well, I was thinking of "endpoint" as just the method that was called. But you are right, it would be ncier if we showed the whole uri. I'll do that change tonight :)